libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -688,55 +700,36 @@ bool PbftManager::stateOperations_() {
-      using namespace std::chrono_literals;
-      std::this_thread::sleep_for(50ms);
-      return true;
-    }
+  // Remove old votes
+  vote_mgr_->cleanupVotes(period, round);
 
-    // Node is in sync
-    new_round_in_sync_ = true;
-  }
+  if (auto certified_block = vote_mgr_->getVotesBundle(round, period, 3, TWO_T_PLUS_ONE); certified_block.has_value()) {
+    LOG(log_dg_) << "PBFT block " << certified_block->voted_block_hash << " has enough cert votes";
+    // put pbft block into chain
+    if (pushCertVotedPbftBlockIntoChain_(certified_block->voted_block_hash, std::move(certified_block->votes))) {
+      LOG(log_nf_) << "Write " << certified_block->votes.size() << " cert votes ... in round " << round;
 
-  // Remove old votes
-  vote_mgr_->cleanupVotes(round);
-
-  // CHECK IF WE HAVE RECEIVED 2t+1 CERT VOTES FOR A BLOCK IN OUR CURRENT
-  // ROUND.  IF WE HAVE THEN WE EXECUTE THE BLOCK
-  // ONLY CHECK IF HAVE *NOT* YET EXECUTED THIS ROUND...
-  if (state_ == certify_state && !have_executed_this_round_) {
-    if (auto certified_block = vote_mgr_->getVotesBundle(round, period, 3, TWO_T_PLUS_ONE);
-        certified_block.has_value()) {
-      LOG(log_dg_) << "PBFT block " << certified_block->voted_block_hash << " has enough cert votes";
-      // put pbft block into chain
-      if (pushCertVotedPbftBlockIntoChain_(certified_block->voted_block_hash, std::move(certified_block->votes))) {
-        db_->savePbftMgrStatus(PbftMgrStatus::ExecutedInRound, true);
-        have_executed_this_round_ = true;
-        LOG(log_nf_) << "Write " << certified_block->votes.size() << " cert votes ... in round " << round;
-
-        duration_ = std::chrono::system_clock::now() - now_;
-        auto execute_trxs_in_ms = std::chrono::duration_cast<std::chrono::milliseconds>(duration_).count();
-        LOG(log_dg_) << "PBFT block " << certified_block->voted_block_hash
-                     << " certified and pushed into chain in round " << round << ". Execution time "
-                     << execute_trxs_in_ms << " [ms]";
-        // Restart while loop
-        return true;
-      } else {
-        LOG(log_er_) << "PBFT block " << certified_block->voted_block_hash
-                     << " certified but not pushed into chain in round " << round;
-      }
+      duration_ = std::chrono::system_clock::now() - now_;
+      auto execute_trxs_in_ms = std::chrono::duration_cast<std::chrono::milliseconds>(duration_).count();
+      LOG(log_dg_) << "PBFT block " << certified_block->voted_block_hash << " certified and pushed into chain in round "
+                   << round << ". Execution time " << execute_trxs_in_ms << " [ms]";
+
+      advancePeriod_();
+
+      // Restart while loop  CONCERN: still need to do that?
+      return true;
+    
+    } else {
+      LOG(log_er_) << "PBFT block " << certified_block->voted_block_hash
+                   << " certified but not pushed into chain in round " << round;
+
+      // CONCERN: How would we get here?
```

Comment:
very easily, you saw 2t+1 cert votes but you are missing 1 some data (dag block, tx, reward vote....) so block was not pushed

Reply:
Ok you are quite right @JakubFornadel